src/scripts/main.js

```diff
@@ -1,3 +1,74 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+const table = document.querySelector('table');
+const body = document.querySelector('body');
+const rows = table.rows;
+let rowCount = 4;
+let columnCount = 4;
+const addRowButton = document.querySelector('.append-row');
+const removeRowButton = document.querySelector('.remove-row');
+const addColumnButton = document.querySelector('.append-column');
+const removeColumnButton = document.querySelector('.remove-column');
+
+addRowButton.addEventListener('click', () => {
+  const rowTr = tbody.children[0].cloneNode(true);
+
+  tbody.append(rowTr);
+
+  rowCount++;
+});
+
+removeRowButton.addEventListener('click', () => {
+  tbody.deleteRow(-1);
+
+  rowCount--;
+});
+
+addColumnButton.addEventListener('click', () => {
+  for (let i = 0; i < rows.length; i++) {
+    rows[i].insertCell(-1);
+  }
+
+  columnCount++;
+});
+
+removeColumnButton.addEventListener('click', () => {
+  for (let i = 0; i < rows.length; i++) {
+    rows[i].deleteCell(-1);
+  }
+
+  columnCount--;
+});
+
+body.addEventListener('click', () => {
```

Comment:
The event listener on the body element might not be the best approach for handling button state updates. Consider updating button states immediately after row or column operations to ensure the UI reflects the current state accurately.